### PR TITLE
adds backend controllers and jbuilder views for contests

### PR DESCRIPTION
**Pull request by LlanddewiLovesYou**
We added controllers and jbuilder views for contests. yay!

### CODE REVIEW

File: test/controllers/contests_controller_test.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class ContestsControllerTest < ActionDispatch::IntegrationTest
+  # test "the truth" do
+  #   assert true
+  # end
+end
```
Comment (Bhammy) [inline test/controllers/contests_controller_test.rb]: Can you explain what this file is?

File: app/controllers/contests_controller.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    if @contest
+      render :show
+    else
+      render json: @contest.errors.full_messages, status: 404
+    end
+  end
+
+  def index
+
+  end
```
Comment (Bhammy) [inline app/controllers/contests_controller.rb]: I'd comment this out or finish the method instead of leaving it as a placeholder.

**Review by Bhammy — CHANGES_REQUESTED** [review]

File: test/controllers/contests_controller_test.rb (hunk 1 above)
Comment (matthaws) [reply test/controllers/contests_controller_test.rb]: Ah. Ian used rails to generate the controller, but that does automatically create a test file. We will remove.

File: app/controllers/contests_controller.rb (hunk 2 above)
Comment (matthaws) [reply app/controllers/contests_controller.rb]: Yes, sir!!

**Review by Bhammy — APPROVED** [review]

**Pull request merged**